Automattic/vip
# Mydumper integration

## Description

This PR adds support for mydumper-based SQL dumps for almost all commands that interact with SQL dumps.

Those commands are:
* `vip dev-env sync sql`
* `vip dev-env import sql`
* `vip search-replace`

What's currently not supported is:
* `vip import sql` - the post-processing involved is included in this PR, but this also requires a back-end change so that  the endpoint it's communicating with can read a `mydumper` file. For now, it's best to use the [datasync feature](https://docs.wpvip.com/databases/data-sync/).

## Pull request checklist


## New release checklist


## Steps to Test

For testing `vip dev-env sync`:

1. Check out PR.
2. Run `npm run build:watch`
3. Ensure that you have a site ready that's part of the mydumper testing group. See `p6jPRI-8tH` shortlink on how to make a site be part of one.
4. Create a new dev-env by running `./dist/bin/vip-dev-env-create.js @0000.production --slug=0000-production` where `0000` is your testing site id. I also suggest that you add phpmyadmin for validation purposes
5. Then, start the dev environment - `./dist/bin/vip-dev-env-start.js @0000.production --slug=0000-production`
6. Run `./dist/bin/vip-dev-env-sync-sql.js @0000.production --slug=0000-production` and your local dev environment should now have its database ready - with all the search and replace working.

For testing `vip dev-env import sql`:

1. Do steps 1 to 5 of testing `vip dev-env sync`.
2. Run `./dist/bin/vip-dev-env-import-sql.js <path-to-mydumper-sql-file> --slug=0000-production`
3. You should see warnings about running search-replace, and the import will partially succeed with failures at creating the admin user.
4. Re-run `./dist/bin/vip-dev-env-import-sql.js <path-to-mydumper-sql-file> --slug=0000-production --search-replace="from,to"` as per its recommendation.
5. Database should import properly this time and the dev-env at the slug `0000-production` will work, just like a mysqldump-based file.

For testing `vip search-replace`:

1. Do steps 1 to 5 of testing `vip dev-env sync`
2. Run `./dist/bin/vip-search-replace.js <mydumper-file> --search-replace="some.from.domain,some.to.domain" --output="result.sql"`. The resulting file should have metadata headers i.e. `-- some_db_wp_site.00000.sql -1` with a -1 at the back. It will also properly replace the values listed in search-replace.
